### ISSUE TITLE
alternateSign function

### ISSUE THREAD

**Issue by romcar** [JS-Challenges/recursion-prompts#57]
Hello, 
 I'm working on the recursion prompts and noticed that the test being done on alternateSign are not working according to specifications.

```// 35. Alternate the numbers in an array between positive and negative regardless of
// their original sign. The first number in the index always needs to be positive.
// alternateSign([2,7,8,3,1,4]) // [2,-7,8,-3,1,-4]
// alternateSign([-2,-7,8,3,-1,4]) // [2,-7,8,-3,1,-4]
```
After completing the function I noticed that the first element being positive gave off an error. I adjusted the functions code to make the first element in the array negative and then alternate from there. I guess  I just wanted to let someone who made it know.

Regards, 
Erwin

**Comment by romcar** [comment]
nvm I think I see what I did

**Comment by romcar** [comment]
Well here'smy code: 

```var alternateSign = function(array) {
  let temp = array.slice();
  let element = temp.shift();

  if(array.length === 0){
    return [];
  }

  //if array length is odd make the element positive 
  if(array.length%2 !== 0){
    //is the number positive 
    return [element > 0 ? 0-element : element].concat(alternateSign(temp));
  } else {
    //if the array length is even, make the element positive

    //is the element negative
    return [element < 0 ? 0-element: element].concat(alternateSign(temp));
  }
  
  return temp;
  /*Almost gave up on this one*/is even make the element positive

    //is the element negative
    return [element < 0 ? 0-element: element].concat(alternateSign(temp));
  }
  
  return temp;
  /*Almost gave up on this one*/
};```


I checked the chai code and it tests to see if the array starts with a positive number. With my code, as the problem recursively flows it determines the sign of the element in the array according to the length of the array. Right now my arrays come out starting with negative values then continuing alternation from there.

 The second if statement (array.length) can be switched to === to make the array being with a positive number. But the test will fail.

**Comment by mybrainishuge** [comment]
Is this your corrected code or the code that returned the false positive? @romcar

**Comment by romcar** [comment]
It was corrected but returned an error. So I changed it back.

**Comment by mybrainishuge** [comment]
I ran the code you provided above and it returns the correct output. Can you help me produce the false positive you saw? @romcar

**Comment by romcar** [comment]
I guess all this studying is making me see things. I promise it was a negative forward alternation. Sorry to take up your time!

**Comment by romcar** [comment]
Thanks

**Comment by romcar** [comment]
When I type in: 

```alternateSign([3, 3, 4])
-->(3) [-3, 3, -4]```

I ran a few console.log test and see that they come out positive but when I type in the above in the console I get a return value of [-3, 3, -4].

**Comment by mybrainishuge** [comment]
For an input of `[3, 3, 4]`, the output should be `[3, -3, 4]`. The numbers returned all even indices should be positive. The numbers returned in all odd indices should be negative. If that's not the output your implementation is returning, then I think you have a bug to fix. :)

This is not an input I included in the tests, so it can't be a bug in the tests. The tests I included are intended to be a guide, but don't account for all possible edge cases. If you want to add tests to account for some edge cases, I'd be thrilled for you to make a PR and I'll review it.
